### PR TITLE
chore: UI fix and improvements

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
- [DET-7394]
- Add sign-in animation

## Test Plan
- Check if new project/workspace/model button won't remain highlighted after closing modal
- Check if User settings dropdown won't remain   highlighted after clicking
- Check if there is a spin after clicking sign in button

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-7394]: https://determinedai.atlassian.net/browse/DET-7394?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/components/DeterminedAuth.tsx [hunk 1]
```diff
@@ -98,7 +98,13 @@ const DeterminedAuth: React.FC<Props> = ({ canceler }: Props) => {
         </p>
       )}
       <Form.Item>
-        <Button disabled={!canSubmit} htmlType="submit" type="primary">Sign In</Button>
+        <Button
+          disabled={!canSubmit}
+          htmlType="submit"
+          loading={!canSubmit}
```
Comment (keita-determined) [inline webui/react/src/components/DeterminedAuth.tsx:104]: Added loading animation

File: webui/react/src/components/NavigationSideBar.tsx [hunk 2]
```diff
@@ -163,11 +163,16 @@ const NavigationSideBar: React.FC = () => {
...
+              <Menu
+                items={[
+                  { key: 'theme-toggle', label: <ThemeToggle /> },
+                  {
+                    key: 'settings',
+                    label: <Link path={paths.settings('account')}>Settings</Link>,
+                  },
+                  { key: 'sign-out', label: <Link path={paths.logout()}>Sign Out</Link> },
+                ]}
+                selectable={false}
```
Comment (keita-determined) [inline webui/react/src/components/NavigationSideBar.tsx:175]: Removed highlight

File: webui/react/src/hooks/useModal/Model/useModalModelCreate.tsx [hunk 3]
```diff
@@ -175,21 +181,19 @@ const useModalModelCreate = (): ModalHooks => {
...
 
   const getModalProps = useCallback((state: ModalState): Partial<ModalFuncProps> => {
-    const { modelName } = state;
     return {
       className: css.base,
       closable: true,
       content: getModalContent(state),
       icon: null,
       maskClosable: true,
-      okButtonProps: { disabled: modelName === '' },
```
Comment (keita-determined) [inline webui/react/src/hooks/useModal/Model/useModalModelCreate.tsx:192]: Somewhat this caused highlight issue, so i removed it and replace it with `Form` to validate

**Review by hkang1 — APPROVED** [review]
This looks and works great! Thanks for making the changes.

**Pull request merged**